embedded/embedded-api/src/main/java/org/mule/runtime/module/embedded/internal/classloading/EnumerationAdapter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.embedded.internal.classloading;
+
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.List;
+
+/**
+ * Creates an {@link Enumeration} containing all the items in a {@link Collection}
+ */
+// TODO MULE-11882 - Consolidate classloading isolation
+public class EnumerationAdapter<T> implements Enumeration<T> {
```

Comment:
Same as before. There's one in the artifact module, but we don't want the client to also depend on that jar.